### DISCUSSION TITLE
pass contextual data to child spans.

### DISCUSSION THREAD

**Discussion by Saurabh-Evertz** [open-telemetry/opentelemetry-cpp#2414] [Q&A]
Hi,
Is there anyway to pass contextual data from parent span to all it's child spans automatically? I read about baggage to achieve something like this but didn't find any specific example for this. Could someone please share an example for this?

**Comment by Falmarri** [comment]
I'm in this same boat right now, looking for examples for using baggage. Three are a couple unit tests, but those don't seem like they're actually how this is intended to be used.
